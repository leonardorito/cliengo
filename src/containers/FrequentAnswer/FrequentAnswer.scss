.FrequentAnswer {
	h1 {
		margin-bottom: 12px;
	}
	
	.row {
		margin-top: 15px;
	}
}