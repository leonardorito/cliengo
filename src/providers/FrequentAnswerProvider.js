import React, { createContext, useState, useEffect } from "react";
import cardData from "../constants/cards-data";
import { ANSWERS } from "../constants/local-storage";

export const FrequentAnswerProvider = createContext();

export const FrequentAnswerContext = ({ children }) => {
  const cards = cardData;

  const [answersList, setAnswersList] = useState(cards);

  const changeActive = (answer) => {
    const list = answersList;
    const answerIndex = list.findIndex((el) => el.id === answer.id);
    list[answerIndex] = {
      ...list[answerIndex],
			active: !answer.active,
			description: answer.description
    };
    setAnswersList(list);
    localStorage.setItem(ANSWERS, JSON.stringify(list));
  };

  useEffect(() => {
    if (!localStorage.getItem(ANSWERS)) {
      const answers = JSON.stringify(answersList);
      localStorage.setItem(ANSWERS, answers);
      return;
    }
    const answers = localStorage.getItem(ANSWERS);
    setAnswersList(JSON.parse(answers));
  }, []);
  return (
    <FrequentAnswerProvider.Provider value={{ answersList, changeActive }}>
      {children}
    </FrequentAnswerProvider.Provider>
  );
};
