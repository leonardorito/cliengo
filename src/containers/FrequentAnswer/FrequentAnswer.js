import React, { useContext, useState } from "react";
import "./FrequentAnswer.scss";
import Card from "../../components/Card/Card";
import { FrequentAnswerProvider } from "../../providers/FrequentAnswerProvider";
import Modal from "../../components/Modal/Modal";
import ModalAnswerContent from "../../components/ModalAnswerContent/ModalAnswerContent";

const FrequentAnswer = () => {
	const { answersList, changeActive } = useContext(FrequentAnswerProvider);
	const [showModal, setShowModal] = useState(false);

	const handleModal = () => {
		setShowModal(!showModal);
	}

	const handleActive = (answer) => {
		changeActive(answer);
	}
  return (
    <div className="FrequentAnswer">
      <h1 className="text-center mt-0">Respuestas frecuentes</h1>
      <div>
        <p>
          <b>
            Cliengo detecta las preguntas más frecuentes que hacen los clientes.
          </b>
        </p>
        <p>
          Optimiza el diálogo con ellos, configurando respuestas a estas
          preguntas desde aquí.
        </p>
        <p>
          <a href="#">Cómo configurar respuestas automáticas</a>
        </p>
      </div>
      <div className="row">
        {answersList &&
          answersList.map((el) => (
            <div key={el.id} onClick={handleModal} className="col">
              <Card data={el} changedActive={() => handleActive(el)} />
            </div>
          ))}
      </div>
      <Modal show={showModal} modalClosed={handleModal}>
        <ModalAnswerContent />
      </Modal>
    </div>
  );
};

export default FrequentAnswer;
