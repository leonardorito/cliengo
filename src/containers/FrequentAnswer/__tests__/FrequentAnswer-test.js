import React from "react";
import cardData from "../../../constants/cards-data";
import { FrequentAnswerProvider } from "../../../providers/FrequentAnswerProvider";
import FrequentAnswer from "../FrequentAnswer";
import renderer from 'react-test-renderer';

it('Frequent answer should render', () => {
  const tree = renderer
    .create(
			<FrequentAnswerProvider.Provider value={{cardData}}>
				<FrequentAnswer />
			</FrequentAnswerProvider.Provider>
		)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
