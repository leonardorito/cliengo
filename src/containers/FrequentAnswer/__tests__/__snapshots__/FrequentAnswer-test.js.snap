// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Frequent answer should render 1`] = `
<div
  className="FrequentAnswer"
>
  <h1
    className="text-center mt-0"
  >
    Respuestas frecuentes
  </h1>
  <div>
    <p>
      <b>
        Cliengo detecta las preguntas más frecuentes que hacen los clientes.
      </b>
    </p>
    <p>
      Optimiza el diálogo con ellos, configurando respuestas a estas preguntas desde aquí.
    </p>
    <p>
      <a
        href="#"
      >
        Cómo configurar respuestas automáticas
      </a>
    </p>
  </div>
  <div
    className="row"
  />
</div>
`;
